internals/secrethub/audit.go

```diff
@@ -63,13 +90,34 @@ func (cmd *AuditCommand) run() error {
 		return err
 	}
 
-	tabWriter := tabwriter.NewWriter(cmd.io.Stdout(), 0, 4, 4, ' ', 0)
-	header := strings.Join(auditTable.header(), "\t") + "\n"
-	fmt.Fprint(tabWriter, header)
+	var formatter tableFormatter
+	if cmd.json {
+		formatter = newJSONFormatter(auditTable.header())
+	} else {
+		terminalWidth, err := cmd.terminalWidth(int(os.Stdout.Fd()))
+		if err != nil {
+			terminalWidth = defaultTerminalWidth
+		}
+		formatter = newColumnFormatter(terminalWidth, auditTable.columns())
+	}
+
+	paginatedWriter, err := cmd.newPaginatedWriter(os.Stdout)
+	if err == errPagerNotFound {
+		paginatedWriter = newFallbackPager(os.Stdout)
+	} else if err != nil {
+		return err
+	}
+	defer paginatedWriter.Close()
+
+	if formatter.printHeader() {
+		header, err := formatter.formatRow(auditTable.header())
+		if err != nil {
+			return err
+		}
+		fmt.Fprintln(paginatedWriter, header)
+	}
```

Comment:
This printing logic is exactly the kind of stuff I'd expect in a tabled formatter or a tabled writer or something like that. It's a bit weird that we're abstracting all the formatting and printing logic away to formatters and pagers, but then end up putting custom logic in this function.

Reply:
I agree, I think the best solution would be to make `formatters` write the output themeselves.